File: medium/3612_process_string_with_special_operations_i/solution.cpp
#include <string>

class Solution {
public:
  std::string processStr(std::string s) {
    std::string result;

    for (const char &ch : s) {
      if (ch == '*') {
        if (!result.empty())
          result.pop_back();
      } else if (ch == '#')
        result += result;
      else if (ch == '%')
        std::reverse(result.begin(), result.end());
      else
        result += ch;
    }

    return result;
  }
};

File: medium/3612_process_string_with_special_operations_i/solution.py
class Solution:
    def processStr(self, s: str) -> str:
        result = []

        for c in s:
            if c == "*":
                if len(result):
                    result.pop()
            elif c == "#":
                result.extend(result)
            elif c == "%":
                result.reverse()
            else:
                result.append(c)

        return "".join(result)
